### DISCUSSION TITLE
Migration of installed RKE2 to a different set of IP addresses

### DISCUSSION THREAD

**Discussion by pgagarinov** [rancher/rke2#2340] [Q&A] [answered]
Is it possible to migrate an installed RKE2 cluster to a different set IPs (in a different subnet)?

**Comment by Martin-Weiss** [comment, marked as answer]
In which scenario do you need this?

Why not deploy a new cluster with new VMs / new IPs / new Firewall settings / new Load Balancer configs / new DNS entries etc. in parallel and finally move the workload..

**Reply by pgagarinov** [reply]
I'm experimenting with RKE2 for PoC and prefer to install it in VMs (Proxmox) that are in a subnet that is directly accessible from my workstation. Once the configuration is done I would prefer to assign a different virtual NIC to each VM and all nodes benind a virtual firewall/loadbalancer to make things more realistic. Yes, installing from scratch is a option but migration to new ips would be easier option (if available) given that I already have VMs and everything installed and configured.

**Reply by Martin-Weiss** [reply]
Not sure if I fully understand - you basically want to access the system via a secondary IP address. This should not be a problem and should not require IP changes.. 

What exactly do you want to access on the RKE2 system - I assume the K8S API and some Ingress - and you need a load balancer in front for that, anyway.. so adding an other load balancer VIP / changing the kubeconfig to point to the other VIP and adding that VIP to additional SANs should do the trick..

**Reply by pgagarinov** [reply]
>Not sure if I fully understand - you basically want to access the system via a secondary IP address. >This should not be a problem and should not require IP changes..

I want to move an existing RKE2 cluster to a different subnet where each node has a different IP. 
192.168.2.10 -> 10.111.33.2
192.168.2.11 -> 10.111.33.3
etc.

I think we are deviating from my original question a bit. It is not about the way to access RKE2, I mentioned the firewall etc just to give you the context (as you requested) explaining why I need to change IPs - because in a different subnet where I would like to move RKE2 to I have cannot use the same IP addresses, that is all. 

As for to load balancer, accessing the cluster via a different IP - I perfectly understand that it is possible via just changing IP and port in kubeconfig. That is not what my question was about. 

Based on your answer I see that it is not possible to change IPs for already configured cluster (please correct me if I'm wrong).

**Reply by Martin-Weiss** [reply]
I believe it might never have been tested to do an IP change without re-install. 

It is well known that changing CNI / CNI CIDRs is not supported after deployment - sometimes things work after changing for some time but then after a patch / upgrade or other change problems show up. Changing the node-names / node-IP addresses might have similar impact.

It might be a good exercise to try it out - shut down rke2 on all nodes, change the IP on the nodes,  start rke2 and see what happens.. - in case the names are used and DNS is working - chances are good that something works but I believe there are several places where also IPs are used and one might realize problems some time later during updates/upgrades or other config changes....

**Reply by pgagarinov** [reply]
Got it, thanks!
